### PR TITLE
Document nn.Sequential.insert and add nn.Sequential.remove

### PR DESCRIPTION
**Pull request by eliemichel**
There was no mention of `insert` it in the documentation and `remove` can be useful!

### CODE REVIEW

**Comment by clementfarabet** [comment]
Cool. A remove(idx) would be nice too.

**Comment by eliemichel** [comment]
Indeed!
Working on it.

**Comment by eliemichel** [comment]
@clementfarabet Did you mean writing the function too? Cause I cannot find it. (Not a pb, just to be sure)

**Comment by clementfarabet** [comment]
yes, I meant to add it! these functions are super useful.

**Comment by eliemichel** [comment]
@clementfarabet ✔

File: Sequential.lua [hunk 1]
```diff
@@ -23,6 +23,16 @@ function Sequential:insert(module, index)
    self.gradInput = self.modules[1].gradInput
 end
 
+function Sequential:remove(index)
+   index = index or #self.modules
+   if index > #self.modules or index < 1 then
+      error"index out of range"
+   end
+   table.remove(self.modules, index)
+   self.output = self.modules[#self.modules].output
```
Comment (fmassa) [inline Sequential.lua]: Don't we need to make the input of the (old) `index+1` module be the output of the `index-1` module as well ? Same thing for the gradInput.

Comment (fmassa) [inline Sequential.lua]: Nevermind, it should work as is. `updateOutput` and `updateGradInput` take care of what I said earlier. My bad !

Comment (eliemichel) [inline Sequential.lua]: @fmassa As the function `insert` does not do such thing, I guessed I didn't had to do it here.

File: doc/containers.md [hunk 2]
```diff
@@ -51,6 +51,47 @@ which gives the output:
...
+> model
+nn.Sequential {
+  [input -> (1) -> (2) -> output]
+  (1): nn.Linear(10 -> 20)
+  (2): nn.Linear(20 -> 30)
+}
+```
+
+
+<a name="nn.Sequential.remove"/>
```
Comment (soumith) [inline doc/containers.md]: this tag here has to be nn.Sequential.insert

**Comment by soumith** [comment]
this is good to go except for a small tag in the documentation. commented there.

**Comment by soumith** [comment]
to not delay this further, I'm merging it in and manually changing the doc bug.

**Comment by eliemichel** [comment]
Sorry, a copy-paste error I guess…
Thanks for fixing+merging!

**Pull request merged**